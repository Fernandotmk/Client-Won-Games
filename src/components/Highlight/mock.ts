import { HighlightProps } from '.'

export default {
  title: 'Read Dead is back',
  subtitle: 'Come see Johns new adventures',
  backgroundImage: '/img/red-dead-img.jpg',
  buttonLabel: 'Buy now',
  buttonLink: '/rdr2',
  floatImage: '/img/red-dead-float.png',
  alignment: 'right'
} as HighlightProps
