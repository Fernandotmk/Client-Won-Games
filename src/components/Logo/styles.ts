import styled, { css } from 'styled-components'
// importando a tipagem feita no index
import { LogoProps } from '.'
import media from 'styled-media-query'

const wrapperModifiers = {
  normal: () => css`
    width: 11rem;
    height: 3.3rem;
  `,
  large: () => css`
    width: 20rem;
    height: 5.9rem;
  `,
  hideOnMobile: () => css`
    ${media.lessThan('medium')`
      width: 5.8rem;
      height: 4.5rem;
      
      svg {
        height: 4.5rem;
        pointer-events: none;
      }
      .text {
        display: none;
      }
    `}
  `
}

// informando que a função poderá receber os tipos da tipagem LogoProps
export const Wrapper = styled.div<LogoProps>`
  ${({ theme, color, size, hideOnMobile }) => css`
    // para não colocar acima color = 'white'
    // pois já está definido no index o padrao white
    // entao é colocado ! do lado de color
    color: ${theme.colors[color!]};

    // se o size existe ( usa dupla negação !! vira)
    ${!!size && wrapperModifiers[size]}

    // se estiver ativado como mobile
    ${!!hideOnMobile && wrapperModifiers.hideOnMobile}
  `}
`
