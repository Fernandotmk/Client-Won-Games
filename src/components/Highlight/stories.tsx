import { ComponentStory, ComponentMeta } from '@storybook/react'
import Highlight from '.'
import item from './mock'

export default {
  title: 'Highlight',
  component: Highlight,
  args: { ...item }
} as ComponentMeta<typeof Highlight>

export const Template: ComponentStory<typeof Highlight> = (args) => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Highlight {...args} />
  </div>
)

export const WithImage = Template.bind({})
WithImage.args = {
  floatImage: '/img/red-dead-float.png'
}
