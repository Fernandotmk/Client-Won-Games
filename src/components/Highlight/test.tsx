import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Highlight from '.'
import * as S from './styles'

const props = {
  title: 'Heading 1',
  subtitle: 'Heading 2',
  backgroundImage: '/img/red-dead-img.jpg',
  buttonLabel: 'Buy Now',
  buttonLink: '/rdr2',
  floatImage: '/img/red-dead-float.png'
}

describe('<Highlight />', () => {
  it('should render the heading', () => {
    // são três estapas do teste:
    // 1- renderizar o componente 'render' (cria um dom)
    // 2- seleconar o elemento a ser testado 'scren' (queries) - getByLabel...
    // 3- expect/assertion - comparação / análise (espero que renderize a logo branca)

    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(
      screen.getByRole('heading', { name: /Heading 1/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /Heading 2/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /buy now/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render background image', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(container.firstChild).toHaveStyle({
      backgroundImage: `url(${props.backgroundImage})`
    })
  })

  it('should render float image', () => {
    renderWithTheme(<Highlight {...props} floatImage="/float-image.png" />)

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      '/float-image.png'
    )
  })

  it('should render align right by default ', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    // espera que o content esteja na esquerda
    // como no grid       "'floatimage content'"
    expect(container.firstChild).toHaveStyleRule(
      'grid-template-areas',
      "'floatimage content'"
    )

    // quando o Content tiver na direita o texto tem que estar também na direita
    expect(container.firstChild).toHaveStyleRule('text-align', 'right', {
      modifier: `${S.Content}`
    })
  })

  it('should render align left ', () => {
    const { container } = renderWithTheme(
      <Highlight {...props} alignment="left" />
    )

    // espera que o content esteja na direita
    // como no grid       "'content floatimage '"
    expect(container.firstChild).toHaveStyleRule(
      'grid-template-areas',
      "'content floatimage'"
    )

    expect(container.firstChild).toHaveStyleRule('text-align', 'left', {
      modifier: `${S.Content}`
    })
  })
})
