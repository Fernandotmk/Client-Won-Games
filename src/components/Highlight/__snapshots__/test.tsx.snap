// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render the heading 1`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c6:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  position: relative;
  background-image: url(/img/red-dead-img.jpg);
  background-position: center center;
  background-size: cover;
  height: 23rem;
  display: grid;
  grid-template-areas: 'floatimage content';
  grid-template-columns: 1.3fr 2fr;
}

.c0::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.6);
}

.c0 .c2 {
  text-align: right;
}

.c1 {
  grid-area: floatimage;
  z-index: 10;
  max-height: 23rem;
  max-width: 100%;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
}

.c3 {
  grid-area: content;
  z-index: 10;
  padding: 1.6rem;
}

.c4 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c5 {
  font-size: 1.4rem;
  font-weight: 300;
  color: #FAFAFA;
  margin-bottom: 2.4rem;
}

@media (min-width:768px) {
  .c0 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c1 {
    max-height: 32rem;
  }
}

@media (min-width:768px) {
  .c3 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 1.8rem;
  }
}

<section
  class="c0"
>
  <img
    alt="Heading 1"
    class="c1"
    src="/img/red-dead-float.png"
  />
  <div
    class="c2 c3"
  >
    <h2
      class="c4"
    >
      Heading 1
    </h2>
    <h3
      class="c5"
    >
      Heading 2
    </h3>
    <a
      class="c6"
      href="/rdr2"
    >
      <span>
        Buy Now
      </span>
    </a>
  </div>
</section>
`;
